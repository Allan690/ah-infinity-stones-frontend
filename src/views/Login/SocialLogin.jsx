import React, { PureComponent } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import GoogleLogin from 'react-google-login';
import { socialLoginRequest } from '../../redux/actions/socialLoginActions';
import socialConfigs from '../../helpers/socialAuthHelpers';
import firebaseAuth from '../../config/firebase';

export class SocialLogin extends PureComponent {
  static propTypes = {
    socialLoginRequest: PropTypes.func.isRequired,
    success: PropTypes.oneOfType([PropTypes.string, PropTypes.bool]),
  };

  static defaultProps = {
    success: null,
  };

  handleTwitter = async (event) => {
    event.preventDefault();
    const response = await firebaseAuth.twitterAuth();
    const { socialLoginRequest: login } = this.props;
    const info = socialConfigs.twitter(response.credential);
    login(info);
  }

  handleFacebookLogin = async (event) => {
    event.preventDefault();

    const response = await firebaseAuth.facebookAuth();
    const { socialLoginRequest: login } = this.props;
    const info = socialConfigs.facebook(response.credential);
    login(info);
  }

  handleGoogleLogin = (response) => {
    const info = socialConfigs.google(response);
    const { socialLoginRequest: login } = this.props;
    login(info);
  };

  handleRedirect = () => { window.location.href = '/'; };

  render() {
    const { success } = this.props;
    return (
      <div className="social-auth">
        { success ? this.handleRedirect() : null }
        <button
          type="button"
          onClick={this.handleFacebookLogin}
          className="kep-login-facebook "
        >
          Facebook
        </button>
        <GoogleLogin
          clientId="31802512181-sh97b417crdhtjtuvabt05chc1e5tn9s.apps.googleusercontent.com"
          buttonText="Google"
          onSuccess={this.handleGoogleLogin}
          onFailure={this.handleGoogleLogin}
        />
        <button
          type="button"
          onClick={this.handleTwitter}
          id="twitter"
        >
          Twitter
        </button>
      </div>
    );
  }
}

const mapStateToProps = (state) => {
  const { socialLoginReducer: { success } } = state;
  return { success };
};

const mapDispatchToProps = { socialLoginRequest };

export default connect(mapStateToProps, mapDispatchToProps)(SocialLogin);
