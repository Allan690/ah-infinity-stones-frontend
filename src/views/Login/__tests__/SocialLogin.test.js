import React from 'react';
import { SocialLogin } from '../SocialLogin';

describe('Social Login Component', () => {
  const props = {
    socialLoginRequest: jest.fn(),
  };
  const wrapper = shallow(<SocialLogin {...props} />);

  it('Should render properly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
