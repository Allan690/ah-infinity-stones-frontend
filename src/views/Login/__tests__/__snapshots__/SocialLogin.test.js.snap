// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Social Login Component Should render properly 1`] = `
<div
  className="social-auth"
>
  <button
    className="kep-login-facebook "
    onClick={[Function]}
    type="button"
  >
    Facebook
  </button>
  <e
    accessType="online"
    buttonText="Google"
    clientId="31802512181-sh97b417crdhtjtuvabt05chc1e5tn9s.apps.googleusercontent.com"
    cookiePolicy="single_host_origin"
    disabledStyle={
      Object {
        "opacity": 0.6,
      }
    }
    fetchBasicProfile={true}
    icon={true}
    isSignedIn={false}
    jsSrc="https://apis.google.com/js/api.js"
    onFailure={[Function]}
    onRequest={[Function]}
    onSuccess={[Function]}
    prompt=""
    scope="profile email"
    tag="button"
    theme="light"
    type="button"
    uxMode="popup"
  />
  <button
    id="twitter"
    onClick={[Function]}
    type="button"
  >
    Twitter
  </button>
</div>
`;
